File: languages/go/entity/identity/number_attribute.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type NumberAttribute struct {
	Attribute
}

func NewNumberAttribute(key string, value int) NumberAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return NumberAttribute{
		Attribute: a,
	}
}

func NewNumberAttributeFromProto(s *proto.NumberAttribute) NumberAttribute {
	if s == nil {
		return NumberAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return NumberAttribute{
		Attribute: a,
	}
}

func (s NumberAttribute) ToProto() *proto.NumberAttribute {
	return &proto.NumberAttribute{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(int64),
	}
}

File: languages/js/src/entity/identity/number_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";
import { Attribute } from "./attribute";

export class NumberAttribute extends Attribute<number> {
  public toProto(): identityEntitiesProto.NumberAttribute {
    return identityEntitiesProto.NumberAttribute.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(r: identityEntitiesProto.NumberAttribute): NumberAttribute {
    return new NumberAttribute(r.id, r.value);
  }
}
